### DISCUSSION TITLE
How to use native C/C++ code with Kha?

### DISCUSSION THREAD

**Discussion by Sanva** [Kode/Kha#1258] [Q&A] [answered]
Hi!

I'm trying to advance in my understanding of Kha cross-platform build system, basically to answer the question How to use native C/C++ code with Kha in Android and Linux?

I'm aware of the existence of

- `haxebullet` — but it only supports (JavaScript and) Hashlink, which isn't the main native target of Kha.
- `haxerecast` — looks outdated, doesn't compile to Linux nor Android, at least from Linux.
- `share_kha` — selected as the main example to create native library bindings for Kha in the wiki, but again it is outdated, it doesn't compile for Android (at least from Linux) and I would be much more interested in a cross-platform native library.

Anything that should be working nowadays?

**Comment by RobDangerous** [comment, marked as answer]
Hi,

I created a sample at https://github.com/Kha-Samples/NativeLibrary
The basic procedure is to put a Kinc project with a kincfile somewhere, then add it via an addLibrary call in your khafile. Then you can use everything that hxcpp has to offer in terms of integrating Haxe with C/C++.
You can see how Kinc projects and kincfiles work at https://github.com/Kinc-Samples - it's very similar to Kha and its khafiles.

PS: Super thanks for copying your question to GitHub discussions.
